lib/media/media_source_engine.js

```diff
@@ -936,18 +942,28 @@ shaka.media.MediaSourceEngine = class {
         if (this.firstVideoTimestamp_ == null &&
             contentType == ContentType.VIDEO) {
           this.firstVideoTimestamp_ = timestamp;
+          this.firstVideoReferenceStartTime_ = reference.startTime;
           if (this.firstAudioTimestamp_ != null) {
-            const compensation =
-                this.firstVideoTimestamp_ - this.firstAudioTimestamp_;
+            let compensation = 0;
+            if (this.firstVideoReferenceStartTime_ ==
```

Comment:
This is becoming complicated enough it'd be worth a comment, I feel.

```js
// Only apply compensation if video and audio segment sizes match, to avoid
// introducing sync issues.
```

Would something like that be an accurate summary?

Reply:
I'd prefer:

```
// Only apply compensation if video and audio segment startTime
// match, to avoid introducing sync issues.
```